src/pepr/operator/controllers/network/generators/kubeNodes.ts

```diff
@@ -0,0 +1,146 @@
+/**
+ * Copyright 2024 Defense Unicorns
+ * SPDX-License-Identifier: AGPL-3.0-or-later OR LicenseRef-Defense-Unicorns-Commercial
+ */
+
+import { KubernetesListObject } from "@kubernetes/client-node";
+import { V1NetworkPolicyPeer, V1NodeCondition, V1NodeAddress } from "@kubernetes/client-node";
+import { K8s, kind } from "pepr";
+import { Component, setupLogger } from "../../../../logger";
+import { RemoteGenerated } from "../../../crd";
+import { anywhere } from "./anywhere";
+import { UDSConfig } from "../../../../config";
+
+const log = setupLogger(Component.OPERATOR_GENERATORS);
+
+// Maintain a set of all node internal IPs
+const nodeSet = new Set<string>();
+
+/**
+ * Initialize the node targets by fetching the current nodes in the cluster
+ * and populating the nodeSet with their Internal IPs.
+ */
+export async function initAllNodesTarget() {
+  try {
+    const nodes = await fetchKubernetesNodes();
+    nodeSet.clear();
+
+    for (const node of nodes.items) {
+      const ip = getNodeInternalIP(node);
+      if (ip) nodeSet.add(ip);
+    }
+  } catch (err) {
+    log.error("error fetching node IPs:", err);
+  }
+}
+
+/**
+ * Returns the egress CIDRs of all known nodes as network policy peers.
+ * If none are known, defaults to 0.0.0.0/0 and logs a warning.
+ */
+export function nodeCIDRs(): V1NetworkPolicyPeer[] {
+  const policies = buildNodePolicies([...nodeSet]);
+  if (policies.length > 0) return policies;
+
+  log.warn("Unable to get API server CIDR, defaulting to 0.0.0.0/0");
```

Comment:
```suggestion
  log.warn("Unable to get Node CIDRs, defaulting to 0.0.0.0/0");
```
(feel free to modify, but this should be nodes rather than api server)